File: 0001-two-sum/0001-two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $map = [];
        foreach ($nums as $index => $num) {
            $comp = $target - $num;
            if (isset($map[$comp])) {
                return [$map[$comp], $index];
            }
            $map[$num] = $index;
        }
        return [];
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for idx, num in enumerate(nums):
            comp = target - num
            if comp in map:
                return [map[comp], idx]
            map[num] = idx
        return []
        
        